x-pack/plugins/lens/public/shared_components/axis_title_settings.tsx

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiText,
+  EuiSwitch,
+  EuiSpacer,
+  EuiFieldText,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { AxesSettingsConfig } from '../../common/expressions';
+import { useDebouncedValue } from './';
+type AxesSettingsConfigKeys = keyof AxesSettingsConfig;
+
+export interface AxisTitleSettingsProps {
+  /**
+   * Determines the axis
+   */
+  axis: AxesSettingsConfigKeys;
+  /**
+   * Determines the axis title
+   */
+  axisTitle: string | undefined;
+  /**
+   * Callback to axis title change
+   */
+  updateTitleState: (value: string) => void;
+  /**
+   * Determines if the title visibility switch is on and the input text is disabled
+   */
+  isAxisTitleVisible: boolean;
+  /**
+   * Toggles the axis title visibility
+   */
+  toggleAxisTitleVisibility: (axis: AxesSettingsConfigKeys, checked: boolean) => void;
+}
+
+export const AxisTitleSettings: React.FunctionComponent<AxisTitleSettingsProps> = ({
+  axis,
+  axisTitle,
+  updateTitleState,
+  isAxisTitleVisible,
+  toggleAxisTitleVisibility,
+}) => {
+  const { inputValue: title, handleInputChange: onTitleChange } = useDebouncedValue<string>({
+    value: axisTitle || '',
+    onChange: updateTitleState,
+  });
+  return (
+    <>
+      <EuiFlexGroup gutterSize="s" justifyContent="spaceBetween">
+        <EuiFlexItem grow={false}>
+          <EuiText size="xs">
+            <h4>
+              {i18n.translate('xpack.lens.shared.axisNameLabel', {
+                defaultMessage: 'Axis name',
```

Comment:
`Axis name` here and `Axis title` few lines down. I think it's better to pick one only.